addon-test-support/@ember/test-helpers/-internal/helper-hooks.ts

```diff
@@ -0,0 +1,48 @@
+type HookUnregister = {
+  unregister: () => void;
+};
+
+const registeredHooks = new Map<string, Set<Function>>();
+
+/**
+ * Registers a hook function to be run during the invocation of a test helper.
+ *
+ * @public
+ * @param {string} helperName The name of the test helper in which to run the hook.
+ * @param {Function} hook The hook function to run when the test helper is invoked.
+ * @returns {HookUnregister} An object containing an unregister function that will unregister
+ *                           the specific hook registered to the helper.
+ */
+export function registerHook(helperName: string, hook: Function): HookUnregister {
```

Comment:
Can we make the hook function types to be a bit more specific? For example, I'd like to have this indicate that we can return a promise from these hooks.

Reply:
Yep, I think that makes sense.